Sprint 7

vanorl opened the pull request:
Sprint 7

vanorl committed:
Sprint 7

vanorl committed:
Sprint 7

DolgoarshinnykhLG commented on data.py:
```diff
@@ -0,0 +1,20 @@
+BASE_URL = "https://qa-scooter.praktikum-services.ru/api/v1"
```
Можно лучше: url-адреса лучше хранить отдельно, они постоянно переиспользуются. Можно вынести в модуль urls

DolgoarshinnykhLG commented on tests/conftest.py:
```diff
@@ -0,0 +1,16 @@
+from courier_methods import CourierMethods
+from auth_methods import AuthMethods
+
+@pytest.fixture
+def courier_data():
+    return generate_courier()
```
Нужно исправить: фикстуры не занимаются выполнением примитивной логики, они выполняют сложную логику предусловий\постусловий и вычислений. Эти методы необходимо описать в модуле helpers и оттуда вызывать в тестах и методах

DolgoarshinnykhLG commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,57 @@
+            assert response.status_code == 201
+            assert response.json() == {"ok": True}
+
+        with allure.step("Удаляем созданного курьера"):
+            courier_id = AuthMethods.login_courier(courier_data).json().get("id")
+            CourierMethods.delete_courier(courier_id)
```
Необходимо исправить здесь и далее: тесты должны заканчиваться assert-инструкцией. Этот вызов является пост-условием и должен размещаться в фикстурах

DolgoarshinnykhLG commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,57 @@
+
+        with allure.step("Проверяем, что вернулась ошибка 409"):
+            assert duplicate_response.status_code == 409
+            assert "уже используется" in duplicate_response.json().get("message", "")
```
Можно лучше тестах этого модуля: все тестовые данные лучше хранить в модуле data, так их будет легче поддерживать

vanorl committed:
Урлы вынесены в urls
Ожидаемые статус коды и тела ответов вынесены в data
Удалена фикстура courier_data
Добавлены фикстуры для удаления курьера и заказа после теста, теперь тесты заканчиваются ассертами

vanorl committed:
небольшая правка
